app/models/user.rb

```diff
@@ -7,17 +7,21 @@ class User < ApplicationRecord
 
   belongs_to :account
 
-  delegate :org_id, to: :account
+  delegate :org_id, :cert_authenticated?, to: :account
 
   def authorized_to?(access_request)
-    return true if ActiveModel::Type::Boolean.new.cast(Settings.disable_rbac)
+    return true if rbac_disabled?
 
     rbac_permissions.any? do |access|
       Rbac.verify(access.permission, access_request)
     end
   end
 
   def inventory_groups
+    # We should not fetch inventory groups if the RBAC feature is globally disabled
+    # or the request is authenticated via an entitlement certificate.
+    return Rbac::ANY if rbac_disabled? || cert_authenticated?
```

Comment:
Shouldn't we restrict all inventory groups (return empty list) in case of cert auth?